rbac/management/utils.py

```diff
@@ -50,11 +51,11 @@ def get_principal_from_request(request):
     """Obtain principal from the request object."""
     current_user = request.user.username
     qs_user = request.query_params.get(USERNAME_KEY)
+    principal_permission = PrincipalAccessPermission()
```

Comment:
Should we make the class (RoleAccessPermission, PrincipalAccessPermission) into singleton or make that method static so we don't have to initiate it each time?

Reply:
@astrozzc it doesn't look like we can use DRF to do that out of the box, since it's a view-level method which the view takes an instance of: https://www.django-rest-framework.org/api-guide/permissions/#custom-permissions

But I do agree that it's not ideal to have this instantiated on each request. We could consider adding an attribute to the module, to allow for something like:

```python
PRICIPAL_PERMISSION_INSTANCE = PrincipalAccessPermission()
...
...
def get_principal_from_request(request):
    ...
    ...
    if qs_user and not PRICIPAL_PERMISSION_INSTANCE.has_permission(request=request, view=None):
```